src/Microsoft.Azure.SignalR/ServiceHubConnectionContext.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO.Pipelines;
+using System.Security.Claims;
+using System.Threading;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.Connections.Features;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Connections.Features;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.AspNetCore.SignalR.Protocol;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace Microsoft.Azure.SignalR
+{
+    /// <summary>
+    /// ServiceHubConnectionContext is only passed to IUserIdProvider.GetUserId as a paramter.
+    /// Only HttpContext and User properties are available for access.
+    /// Exception will be thrown when other properties are accessed.
+    /// </summary>
+    internal class ServiceHubConnectionContext : HubConnectionContext
+    {
+        public const string UnavailableErrorTemplate = " property is not available in current context of Azure SignalR Service.";
+        public const string ConnectionAbortedUnavailableError = nameof(ConnectionAborted) + UnavailableErrorTemplate;
+        public const string ConnectionIdUnavailableError = nameof(ConnectionId) + UnavailableErrorTemplate;
+        public const string ItemsUnavailableError = nameof(Items) + UnavailableErrorTemplate;
+        public const string ProtocolUnavailableError = nameof(Protocol) + UnavailableErrorTemplate;
+
+        public ServiceHubConnectionContext(HttpContext context)
+            : base(new DummyConnectionContext(context), TimeSpan.MaxValue, NullLoggerFactory.Instance)
+        {
+        }
+
+        public override CancellationToken ConnectionAborted => throw new InvalidOperationException(ConnectionAbortedUnavailableError);

```

Comment:
Why throw here? Most of these properties can just do nothing.

Reply:
Per offline sync, we will leave these exceptions as they are now to prevent users using these properties.